Terminal.Gui/Core.cs

```diff
@@ -2218,6 +2351,8 @@ public static void Shutdown ()
 
 		static void Redraw (View view)
 		{
+			Application.CurrentView = view;
```

Comment:
Should we not restore this value after Redraw?

But generally, this looks like some global state that is prime to be trampled over, I would rather not have it.

What is the rationale for having this?

Reply:
That static variable it's very important for to know what view is being drawn when a Rune is added to the screen. In case it exceeds the screen limits it's not drawn unless the AllowNewLine is setted. I can change it to Auto-Wrap wich I thing it's looks a beter name.

Reply:
> Should we not restore this value after Redraw?

Application.CurrentView is restored at Redraw method of the Application class, View class, Toplevel class and in the Window class. Thus is restored in all most top-levels.